extensions/stackdriver/stackdriver.cc

```diff
@@ -265,6 +268,26 @@ bool StackdriverRootContext::configure(size_t configuration_size) {
   }
 
   if (!logger_ && enableAccessLog()) {
+    std::unordered_map<std::string, std::string> extra_labels;
+    if (config_.has_custom_log_config()) {
+      cleanupExpressions();
+      for (const auto& dimension : config_.custom_log_config().dimensions()) {
+        uint32_t token;
+        if (createExpression(dimension.second, &token) != WasmResult::Ok) {
+          LOG_TRACE(absl::StrCat("Could not create expression for ",
+                                 dimension.second));
+          continue;
+        }
+        std::string value;
+        if (evaluateExpression(token, &value)) {
+          extra_labels[dimension.first] = value;
+          continue;
+        }
+        // Push back expressions that could not be evaluated at config time, to
+        // be evaluated at request time.
```

Comment:
Some resources can be evaluated using node metadata and hence can be done at config time itself.. EX in test itself "source_workload: "'_' + node.metadata['WORKLOAD_NAME']"
    source_workload_namespace: "'_' + node.metadata['NAMESPACE']"" 
Things like above could be evaluated at config time... wdyt? 
No harm in checking at config time to avoid paying cost in request path?

Reply:
Are we sure that per request expression cannot be successfully evaluated at config time?

Reply:
Expressions like "request.protocol" can be evaluated at request time only and not at config time...

Reply:
Yeah, I am just worrying that we cannot make sure every request dependent attribute cannot be evaluated on config time. There are quite a lot of attributes, I don't think we can be certain on each of them.

Reply:
so there are scenarios wherein an expression evaluated at config time is different from request time? then I will delete this code of evaluating at config time...

One of the other concerns I had was about source labels being merged at request time but that should be fine too as below:
Things that can be evaluated at config time like source information, in our logger we add it in base WriteLogEntryRequest and then it gets combined into one set of labels on stackdriver side. So if we evaluate custom "source_workload" at request time, it will get added in log entry labels and then on stackdriver side they will keep log entry labels "source_workload" as per `Optional. Default labels that are added to the labels field of all log entries in entries. If a log entry already has a label with the same key as a label in this parameter, then the log entry's label is not changed. See LogEntry.(https://cloud.google.com/logging/docs/reference/v2/rest/v2/entries/write) `

Reply:
Can we make this explicit if we want to keep it? The problem is that an error does not indicate that the expression is config vs request time. The expression could have some conditional like `has(request.path) ? request.path : ""` that would make it evaluate at config time, which is not what we want.

Reply:
Changed it to just evaluating at request time..